### PR TITLE
Changed the group picker to reflect the updated design

### PR DESCRIPTION
**Pull request by edewit**
Now follows the design more closely

### CODE REVIEW

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/components/group/GroupPickerDialog.tsx [hunk 1]
```diff
@@ -87,36 +91,42 @@ export const GroupPickerDialog = ({
       groups.forEach((group: SelectableGroup) => {
         group.checked = !!selectedRows.find((r) => r.id === group.id);
       });
-      setFiltered(undefined);
-      setFilter("");
-      setFirst(0);
-      setMax(10);
```
[the remark this answers is not in the document]
Comment (edewit) [reply src/components/group/GroupPickerDialog.tsx:93]: Yes

File: src/components/group/GroupPickerDialog.tsx [hunk 2]
```diff
@@ -87,36 +91,42 @@ export const GroupPickerDialog = ({
...
+      setGroups(groups);
     },
-    [groupId]
+    [groupId, filter, first, max]
   );
 
   const isRowDisabled = (row?: GroupRepresentation) => {
-    return !!joinedGroups.find((group) => group.id === row?.id);
+    return !![...joinedGroups, ...(filterGroups || [])].find(
+      (group) => group.id === row?.id
```
[the remark this answers is not in the document]
Comment (edewit) [reply src/components/group/GroupPickerDialog.tsx:101]: Every group has an `id` it's an error in the type information that this field might be optional. It's generated from the java classes and a java `String` can be null

File: src/components/group/GroupPickerDialog.tsx [hunk 3]
```diff
@@ -60,23 +60,27 @@ export const GroupPickerDialog = ({
 
   useFetch(
     async () => {
-      const allGroups = await adminClient.groups.find();
+      let group;
```
[the remark this answers is not in the document]
Comment (edewit) [reply src/components/group/GroupPickerDialog.tsx:63]: no, it should only make one request

File: src/components/group/GroupPickerDialog.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply src/components/group/GroupPickerDialog.tsx:93]: Got it :ok_hand:

File: src/components/group/GroupPickerDialog.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply src/components/group/GroupPickerDialog.tsx:101]: I see, if we are sure that this field is always present then I would say let's update the type information to reflect the real world expectations.

File: src/components/group/GroupPickerDialog.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply src/components/group/GroupPickerDialog.tsx:63]: I see that it doens't do so in this case. But if an `id` is provided to the `GroupPickerDialog` it should fire two requests.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**